[feat] 반려견 정보 저장 기능 구현

green-yoon87 opened the pull request:
### 🔖  Issue Number

close #74

### 📙 작업 내역

> 반려견 정보 저장 기능을 구현한다. 
> 
- [x]  성별 및 중성화 정보 입력 받는 컴포넌트 추가 
- [x] 스크롤 뷰 적용 
- [x] 모든 값을 입력하는 조건에만 버튼 활성화 
- [x] 키워드 버튼 최대 2개까지 선택 가능 - 2개 선택이면 나머지 비활성화 
- [x] Dog, DogInfo Entity 구조 수정 
- [x] 나이 textfield에 숫자 키보드 설정 및 글자수(2자) 제한 

https://github.com/user-attachments/assets/c1ae39f5-e6ca-4504-a44b-8b098ad9cf0d

<img width="240" alt="스크린샷 2024-11-07 오후 3 18 08" src="https://github.com/user-attachments/assets/e0e1b3c1-8530-4fc8-9bc2-1269fc240a38">

- 특이 사항 1 
navigationcontroller의 backbutton 색과 타이틀을 설정하는 코드를 작성했습니다. (mainNavy 색이고 타이틀 없는 버전) 
아래와 같이 사용하시면 됩니다!
```swift
navigationController?.navigationBar.configureBackButton()
```

green-yoon87 committed:
[feat] #74 성별, 중성화 여부 정보 입력받는 컴포넌트 추가

green-yoon87 committed:
[feat]#74 스크롤뷰 추가

se, 16, 16 pro max 테스트했습니다.

green-yoon87 committed:
[feat] #74 강아지 정보 entity에 성별과 중성화 유무 추가

green-yoon87 committed:
[feat] #74 나이를 입력받는 textfield에 숫자키패드로만 입력받을 수 있도록 설정 및 길이 2로 제한

green-yoon87 committed:
[feat] #74 크기, 키워드, 중성화 여부, 성별을 포함한 모든 정보 입력 받기 구현

green-yoon87 committed:
[fix] #74 Stackview를 추가하여 keyword button들을 관리하도록 수정

green-yoon87 committed:
[feat] #74 키워드 버튼 액션 연결 및 버튼 활성화 조건 구현

- 키워드 버튼 최대 2개만 선택될 수 있도록 구현
- 키워드 버튼이 한개 이상 눌리고, name, age가 입력되어야만 제출 버튼 활성화

green-yoon87 committed:
[feat] #74 입력한 정보 전달

green-yoon87 committed:
[feat] #74 Navigation backbutton 색 변경 및 타이틀 제거

kelly-chui commented on SniffMeet/SniffMeet/Source/Auth/CreateProfile/View/ProfileInputViewController.swift:
```diff
@@ -24,181 +28,369 @@ final class ProfileInputViewController: UIViewController, ProfileInputViewable {
+    }
+}
+
+private extension ProfileInputViewController {
+    func configureTextFieldConstraints() {
+        NSLayoutConstraint.activate([
```
진짜 고생하셨습니다. 🫡

pearhyunjin commented on SniffMeet/SniffMeet/Source/Auth/CreateProfile/View/ProfileInputViewController.swift:
```diff
@@ -24,181 +28,369 @@ final class ProfileInputViewController: UIViewController, ProfileInputViewable {
-        static let shyKeywordLabel: String = "소심한"
-        static let independentKeywordLabel: String = "독립적인"
+        static let sexUponIntakeArr: [String] = ["완료", "미완료"]
+        static let sexArr: [String] = [Sex.male.rawValue, Sex.female.rawValue]
+        static let sizeArr: [String] = [Size.small.rawValue, Size.medium.rawValue, Size.big.rawValue]
```
훨씬 깔끔하네요! 👍

pearhyunjin commented on SniffMeet/SniffMeet/Source/Auth/CreateProfile/View/ProfileInputViewController.swift:
```diff
@@ -6,14 +6,18 @@
 //
 
 import UIKit
+import Combine
 
```
ㅎㅎ

green-yoon87 replied:
ㅋㅋㅋㅋㅋㅋㅋ 3번째....  알파벳 책 사줘용

pearhyunjin commented on SniffMeet/SniffMeet/Source/Auth/CreateProfile/View/ProfileCreateViewController.swift:
```diff
@@ -60,6 +61,7 @@ final class ProfileCreateViewController: UIViewController, ProfileCreateViewable
         setDelegate()
         setButtonAction()
         hideKeyboardWhenTappedAround()
+        navigationController?.navigationBar.configureBackButton()
```
👍

kelly-chui commented on SniffMeet/SniffMeet/Source/Auth/CreateProfile/View/ProfileInputViewController.swift:
```diff
@@ -304,11 +309,37 @@ private extension ProfileInputViewController {
+            }
+            button.publisher(event: .touchUpInside).sink{ [weak self] in
+                guard let keywordButtons = self?.keywordButtons else {return}
+                var selectedKeywordButtons: [KeywordButton] = []
+                var unSelectedKeywordButtons: [KeywordButton] = []
```
P5. `set`으로 빼서 외부에서 메소드 외부에서 관리하는 방식은 별루일까요??

green-yoon87 replied:
set으로 뺀다는 게 어떤건지 자세히 설명해주실 수 있나용?
확실히 메서드 외부에서 따로 빼는게 가독성 측면에서 좋아보이네요! 수정하겠습니다.

kelly-chui replied:
Set 타입 말하는거였습니다. 안에 있는 애들 삭제/삽입 편하게요!

kelly-chui reviewed (approved):
정말 고생하셨습니다 ~! 🫡🫡

pearhyunjin reviewed (approved):
고생하셨습니다~!
훨씬 깔끔한 코드가 된 것 같아요 👍

green-yoon87 committed:
[feat] #74 선택된 버튼과 그렇지 않은 버튼 계산 속성으로 빼기, 버튼 활성화 설정하는 함수 추가로 작성

이미 button action에서 어떤 버튼에 선택되어있는지, 그 개수가 파악되기 때문에 그걸 활용할 수 있는 함수를 추가로 작성했습니다.
이렇게 된다면 Boolean 값으로 받아 키워드 버튼이 선택되었는지 그렇지 않은지 확인이 가능해 keyword 버튼 리스트를 순회하지 않아도 됩니다.
